crates/dojo-core/src/interfaces.cairo

```diff
@@ -1,13 +1,17 @@
 use dojo_core::integer::u250;
 use dojo_core::string::ShortString;
 use dojo_core::serde::SpanSerde;
+use dojo_core::auth::systems::Route;
+use starknet::{ClassHash, ContractAddress};
 
 #[abi]
 trait IWorld {
-    fn component(name: ShortString) -> starknet::ClassHash;
-    fn register_component(class_hash: starknet::ClassHash);
-    fn system(name: ShortString) -> starknet::ClassHash;
-    fn register_system(class_hash: starknet::ClassHash);
+    fn initialize(routing: Array<Route>);
```

Comment:
lets use the constructor here?

Reply:
Thanks for the review! Got it, I'll move this to the constuctor 👍